### PR TITLE
chore: update cargo to v0.66

### PR DESCRIPTION
**Pull request by MarcoIeni**
I also update all the other dependencies.

### CODE REVIEW

File: cargo-clone-core/src/lib.rs [hunk 1]
```diff
@@ -159,11 +160,19 @@ fn select_pkg<'a, T>(
 where
     T: Source + 'a,
 {
-    src.update()?;
+    src.invalidate_cache();
```
Comment (MarcoIeni) [inline cargo-clone-core/src/lib.rs:163]: [This](https://github.com/rust-lang/cargo/commit/f12f0256caedaeada2fecd8bc5e93feee7abb5b7) is the commit where the `update` function was removed.
The git commit comment is quite good 👍 
A possible optimisation could be to call the `invalidate_cache` function only once. I think at the moment we call it for every crate we download. What do you think? We could open an issue and implement this in a separate PR.

File: cargo-clone-core/src/lib.rs [hunk 2]
```diff
@@ -159,11 +160,19 @@ fn select_pkg<'a, T>(
...
-    src.query(&dep, &mut |summary| summaries.push(summary))?;
+
+    loop {
+        match src.query(&dep, QueryKind::Exact, &mut |summary| {
+            summaries.push(summary)
+        })? {
+            std::task::Poll::Ready(()) => break,
+            std::task::Poll::Pending => src.block_until_ready()?,
+        }
+    }
```
Comment (MarcoIeni) [inline cargo-clone-core/src/lib.rs:175]: The `query` function returns a `Pool`.
[This](https://github.com/rust-lang/cargo/blob/d95124ee93d28bfe6960602130fd083701a3525d/src/cargo/ops/cargo_add/mod.rs#L525) is how cargo waits for the poll to be ready apparently.

File: cargo-clone-core/src/lib.rs (hunk 1 above)
Comment (JanLikar) [reply cargo-clone-core/src/lib.rs:163]: Yes, let's open a ticket for that.

Comment (JanLikar) [reply cargo-clone-core/src/lib.rs:163]: #56

**Review by JanLikar — APPROVED** [review]

**Comment by JanLikar** [comment]
Thanks a lot Marco, I appreciate it!

**Comment by MarcoIeni** [comment]
Thank you for reviewing it so quickly :)

**Pull request merged**